aea/protocols/dialogue/base.py

```diff
@@ -716,6 +721,8 @@ def reply(
 
         self._update(reply)
 
+        reply.envelope_context = target_message.envelope_context
```

Comment:
Alternatively, we construct a new envelope context. But this is cheaper.

Reply:
I think it's ok to share it, most of the time it should stay immutable and used in reply as was provided.